File: userCode.cpp
///START

#include <stdio.h>
#include <opencv2/highgui.hpp>
#include <string>
#include <iostream>
#include "./bibliothequeCpp/Opencv.h"
#include "./bibliothequeCpp/Exercice.h"
#include "./bibliothequeCpp/Matrice.h"
#include "./bibliothequeCpp/Line.h"
#include "./bibliothequeCpp/Pixel.h"

using namespace std;


int doExercice(vector<Matrice> listPattern, Matrice result, int cv[3]){
    int currentId[3] = ;
    int x = 0;
    int y = 0;
    int CO = 0;
    int a,b= 0;
    int k = y<x<a;
    int k = y\a;

    for(unsigned int i = 0; i < listPattern.size(); i++){
        x = 0;
        y = 0;
/*



*/
        bool equaL = true;
        Matrice matrice = listPattern[i];
        while( y < result.getSize() && equaL){
            
            while( x < result.getSize() && equaL){

                if( ! (matrice.getPixel(x,y).compare(result.getPixel(x,y)))){

                    equaL = false;
                }
                x++;
           
            }
            x=0;
            y++;
           //
            }

             if(equaL){
                return currentId;
            }else{//
                currentId++;
        }

    }
    return -1;
}


int test(vector<Matrice> aaa, Matrice bh){
    float aty = 1;
    int ooo = 0;

}

int test(vector<hjgh> aaa, Matrice hhh){
    float aty = 1;
    int ooo = 0;
    int dx;
    if (dx!=0){

    }
}




std::string testAlgo(std::string nameExercice, int resultat, int nbMatriceResult){
    int tesTest =0;
    Opencv opencv = Opencv(nameExercice);
    opencv.setNumberImageResultat(nbMatriceResult);
    opencv.getNumberImage();
    opencv.extractImage();
    opencv.initSizeImage();

    Exercice exercice = Exercice(resultat, nameExercice);
    Exercice(resultat, nameExercice);
    std::vector<Matrice> listPatternInit = opencv.initExercice();
    Matrice matrice_result = opencv.matriceResult();

    int solution_user = doExercice(listPatternInit, matrice_result);
    return exercice.assertRes(solution_user, resultat);
    
    return 0;
}





int main(int argc, char *argv[])
{
    //sdfsfsf
dqsdfqs
unionu
defaultd
defaultd
fsdfs
int d;
wcwxc
bibliothequeCppq
e
cwx
int z;
z
s
a
s
s







a
s
s
s
a
s
a
z
z
z
w
x
c
}
///END

File: userCode.py

### Ajouté par l'API avant l'envoie à judge0

from bibliothequePython import bib

import cv2 as cv

from bibliothequePython.bib import Matrice, Opencv, Exercice





def testAlgo(nameExercice, resultat, nbMatriceResult):
    opencv = Opencv(nameExercice)
    opencv.setNumberImageResultat(nbMatriceResult)
    opencv.getNumberImage()
    opencv.extractImage()
    opencv.initSizeImage()

    exercice = Exercice(resultat, nameExercice)
    listPatternInit = opencv.initExercice(opencv.sizeImage)
    print("test")
    print(opencv.sizeImage)
    matrice_result = opencv.matriceResult(opencv.sizeImage)

    print("pattern")
    for patern in listPatternInit:
        print(patern)


    solution_user = doExercice(listPatternInit, matrice_result)
    print(solution_user)
    return exercice.assertRes(solution_user, resultat)




###Donnée à l'utilisateur


### FIN Ajouté par l'API avant l'envoie à judge0


#Consigne: Vous disposez d'une image résultat et d'une liste d'image.
#Implémentez la fonction doExercice afin de créer un algoritme capable de trouver l'image d'entrée dans la liste d'image de sortie
#
#
#Donnée: Image a trouver => matrice_result

#        Liste d'images a traiter => listMatrice

#Réponse: vous devez retourner l'id de limage dans la liste (0, 1, 2 ...) correspondant à l'image d'entrée


### Algo crée par l'utilisateur


def doExercice(listMatrice, matriceInputToFind, test):
    currentId = 0
    solutionId = -1
    test[3]=1
    CO = 0
    """   
    
    
    """
#fhgfh
    for matrice in listMatrice:#
        ligneInputPattern = 5

        equal = True
        matrice.toStringPixel()

        for lignePixel in matrice.getMatriceContent():
            colonneInputPattern = 0

            for pixel in lignePixel:

                if not comparePixel(colonneInputPattern, ligneInputPattern, matrice, matriceInputToFind):
                    equal = False

                    if equal:
                        solutionId = currentId


                colonneInputPattern = colonneInputPattern +1

            ligneInputPattern = ligneInputPattern +1

        if equal :
            solutionId = currentId

        currentId = currentId + 1
    ox = 100
    o
    return solutionId

def testAlgo():
    exercice = Exercice(resultat, nameExercice)

    return exercice

def comparePixel(x, y, matriceSource, matriceCible):
    return matriceSource.getPixel(x, y).compare(matriceCible.getPixel(x, y))

### FIN  Algo crée par l'utilisateur

if __name__ == '__main__':
    nameExercice = "comparaisonMatrice"
    resultat = 2
  #  print(testAlgo(nameExercice, resultat, nbMatriceResult))

